### ISSUE TITLE
Declaring a Vault path inside server.ssl.keyStorePassword is not resolved correctly

### ISSUE THREAD

**Issue by dwp-kevinpotgieter** [dwp/queue-triage#162]
TL;DR - Register a converter from `char[]` to `String`

Given config that looks something like:

```yaml
server:
  ssl:
    keyStorePassword: "VAULT(/some/secret/path)"   
```
Vault is correctly able to lookup the value, and return the `char[]` representation of this vault path.

However, when the configuration properties bean binding occurs, this value is incorrectly set on the corresponding configuration POJO. (See the `org.springframework.boot.autoconfigure.web.ServerProperties` for more details.)

 The main reason being, is that the `keyStorePassword` field on the `org.springframework.boot.context.embedded.Ssl` class is a `String` not a `char[]`, and there is no relevant `org.springframework.core.convert.converter.Converter` registered to handle this so the default fallback is to call `.toString()` on the object in question.

**Comment by amcghie** [comment]
@dwp-kevinpotgieter  are you happy for this to be closed?
